dockerfiles/debian-buster-all/Dockerfile

```diff
@@ -0,0 +1,83 @@
+# vim:set ft=dockerfile:
+FROM debian:buster
+ARG DEBIAN_FRONTEND=noninteractive
+
+# See https://github.com/tianon/docker-brew-debian/issues/49 for discussion of the following
+#
+# https://bugs.debian.org/830696 (apt uses gpgv by default in newer releases, rather than gpg)
+RUN set -x \
+	&& apt-get update \
+	&& { \
+		which gpg \
+# prefer gnupg2, to match APT's Recommends
+		|| apt-get install -y --no-install-recommends gnupg2 \
+		|| apt-get install -y --no-install-recommends gnupg \
+	; } \
+# Ubuntu includes "gnupg" (not "gnupg2", but still 2.x), but not dirmngr, and gnupg 2.x requires dirmngr
+# so, if we're not running gnupg 1.x, explicitly install dirmngr too
+	&& { \
+		gpg --version | grep -q '^gpg (GnuPG) 1\.' \
+		|| apt-get install -y --no-install-recommends dirmngr \
+	; } \
+	&& rm -rf /var/lib/apt/lists/*
+
+RUN set -ex; \
+# pub   4096R/ACCC4CF8 2011-10-13 [expires: 2019-07-02]
```

Comment:
OK, so I used a tool (available on Debian, at least) called `pgpdump` to print out details of the packets in this key…

```
root@56a50a6a9a82:~# pgpdump key.pub 
Old: Public Key Packet(tag 6)(525 bytes)
	Ver 4 - new
	Public key creation time - Thu Oct 13 20:19:14 UTC 2011
	Pub alg - RSA Encrypt or Sign(pub 1)
	RSA n(4096 bits) - ...
	RSA e(17 bits) - ...
Old: User ID Packet(tag 13)(28 bytes)
	User ID - PostgreSQL Debian Repository
Old: Signature Packet(tag 2)(573 bytes)
	Ver 4 - new
	Sig type - Positive certification of a User ID and Public Key packet(0x13).
	Pub alg - RSA Encrypt or Sign(pub 1)
	Hash alg - SHA256(hash 8)
	Hashed Sub: key flags(sub 27)(1 bytes)
		Flag - This key may be used to certify other keys
		Flag - This key may be used to sign data
	Hashed Sub: preferred symmetric algorithms(sub 11)(4 bytes)
		Sym alg - AES with 256-bit key(sym 9)
		Sym alg - AES with 192-bit key(sym 8)
		Sym alg - AES with 128-bit key(sym 7)
		Sym alg - CAST5(sym 3)
	Hashed Sub: preferred hash algorithms(sub 21)(4 bytes)
		Hash alg - SHA512(hash 10)
		Hash alg - SHA384(hash 9)
		Hash alg - SHA256(hash 8)
		Hash alg - SHA224(hash 11)
	Hashed Sub: preferred compression algorithms(sub 22)(4 bytes)
		Comp alg - ZLIB <RFC1950>(comp 2)
		Comp alg - BZip2(comp 3)
		Comp alg - ZIP <RFC1951>(comp 1)
		Comp alg - Uncompressed(comp 0)
	Hashed Sub: features(sub 30)(1 bytes)
		Flag - Modification detection (packets 18 and 19)
	Hashed Sub: key server preferences(sub 23)(1 bytes)
		Flag - No-modify
	Hashed Sub: signature creation time(sub 2)(4 bytes)
		Time - Sat Oct 13 08:49:48 UTC 2012
	Hashed Sub: key expiration time(sub 9)(4 bytes)
		Time - Sun Oct 13 08:49:37 UTC 2013
	Sub: issuer key ID(sub 16)(8 bytes)
		Key ID - 0x7FCC7D46ACCC4CF8
	Hash left 2 bytes - 6e 09 
	RSA m^d mod n(4095 bits) - ...
		-> PKCS-1
Old: Signature Packet(tag 2)(573 bytes)
	Ver 4 - new
	Sig type - Positive certification of a User ID and Public Key packet(0x13).
	Pub alg - RSA Encrypt or Sign(pub 1)
	Hash alg - SHA256(hash 8)
	Hashed Sub: key flags(sub 27)(1 bytes)
		Flag - This key may be used to certify other keys
		Flag - This key may be used to sign data
	Hashed Sub: preferred symmetric algorithms(sub 11)(4 bytes)
		Sym alg - AES with 256-bit key(sym 9)
		Sym alg - AES with 192-bit key(sym 8)
		Sym alg - AES with 128-bit key(sym 7)
		Sym alg - CAST5(sym 3)
	Hashed Sub: preferred hash algorithms(sub 21)(4 bytes)
		Hash alg - SHA512(hash 10)
		Hash alg - SHA384(hash 9)
		Hash alg - SHA256(hash 8)
		Hash alg - SHA224(hash 11)
	Hashed Sub: preferred compression algorithms(sub 22)(4 bytes)
		Comp alg - ZLIB <RFC1950>(comp 2)
		Comp alg - BZip2(comp 3)
		Comp alg - ZIP <RFC1951>(comp 1)
		Comp alg - Uncompressed(comp 0)
	Hashed Sub: features(sub 30)(1 bytes)
		Flag - Modification detection (packets 18 and 19)
	Hashed Sub: key server preferences(sub 23)(1 bytes)
		Flag - No-modify
	Hashed Sub: signature creation time(sub 2)(4 bytes)
		Time - Sun Feb 24 19:44:21 UTC 2013
	Hashed Sub: key expiration time(sub 9)(4 bytes)
		Time - Wed Feb 24 19:44:03 UTC 2016
	Sub: issuer key ID(sub 16)(8 bytes)
		Key ID - 0x7FCC7D46ACCC4CF8
	Hash left 2 bytes - 45 31 
	RSA m^d mod n(4095 bits) - ...
		-> PKCS-1
Old: Signature Packet(tag 2)(573 bytes)
	Ver 4 - new
	Sig type - Positive certification of a User ID and Public Key packet(0x13).
	Pub alg - RSA Encrypt or Sign(pub 1)
	Hash alg - SHA256(hash 8)
	Hashed Sub: key flags(sub 27)(1 bytes)
		Flag - This key may be used to certify other keys
		Flag - This key may be used to sign data
	Hashed Sub: preferred symmetric algorithms(sub 11)(4 bytes)
		Sym alg - AES with 256-bit key(sym 9)
		Sym alg - AES with 192-bit key(sym 8)
		Sym alg - AES with 128-bit key(sym 7)
		Sym alg - CAST5(sym 3)
	Hashed Sub: preferred hash algorithms(sub 21)(4 bytes)
		Hash alg - SHA512(hash 10)
		Hash alg - SHA384(hash 9)
		Hash alg - SHA256(hash 8)
		Hash alg - SHA224(hash 11)
	Hashed Sub: preferred compression algorithms(sub 22)(4 bytes)
		Comp alg - ZLIB <RFC1950>(comp 2)
		Comp alg - BZip2(comp 3)
		Comp alg - ZIP <RFC1951>(comp 1)
		Comp alg - Uncompressed(comp 0)
	Hashed Sub: features(sub 30)(1 bytes)
		Flag - Modification detection (packets 18 and 19)
	Hashed Sub: key server preferences(sub 23)(1 bytes)
		Flag - No-modify
	Hashed Sub: signature creation time(sub 2)(4 bytes)
		Time - Wed Jan 29 14:50:01 UTC 2014
	Hashed Sub: key expiration time(sub 9)(4 bytes)
		Time - Tue Jul  2 14:49:19 UTC 2019
	Sub: issuer key ID(sub 16)(8 bytes)
		Key ID - 0x7FCC7D46ACCC4CF8
	Hash left 2 bytes - ce 64 
	RSA m^d mod n(4095 bits) - ...
		-> PKCS-1
Old: Signature Packet(tag 2)(590 bytes)
	Ver 4 - new
	Sig type - Positive certification of a User ID and Public Key packet(0x13).
	Pub alg - RSA Encrypt or Sign(pub 1)
	Hash alg - SHA256(hash 8)
	Hashed Sub: key flags(sub 27)(1 bytes)
		Flag - This key may be used to certify other keys
		Flag - This key may be used to sign data
	Hashed Sub: preferred symmetric algorithms(sub 11)(4 bytes)
		Sym alg - AES with 256-bit key(sym 9)
		Sym alg - AES with 192-bit key(sym 8)
		Sym alg - AES with 128-bit key(sym 7)
		Sym alg - CAST5(sym 3)
	Hashed Sub: preferred hash algorithms(sub 21)(4 bytes)
		Hash alg - SHA512(hash 10)
		Hash alg - SHA384(hash 9)
		Hash alg - SHA256(hash 8)
		Hash alg - SHA224(hash 11)
	Hashed Sub: preferred compression algorithms(sub 22)(4 bytes)
		Comp alg - ZLIB <RFC1950>(comp 2)
		Comp alg - BZip2(comp 3)
		Comp alg - ZIP <RFC1951>(comp 1)
		Comp alg - Uncompressed(comp 0)
	Hashed Sub: features(sub 30)(1 bytes)
		Flag - Modification detection (packets 18 and 19)
	Hashed Sub: key server preferences(sub 23)(1 bytes)
		Flag - No-modify
	Hashed Sub: issuer fingerprint(sub 33)(21 bytes)
	 v4 -	Fingerprint - b9 7b 0a fc aa 1a 47 f0 44 f2 44 a0 7f cc 7d 46 ac cc 4c f8 
	Hashed Sub: signature creation time(sub 2)(4 bytes)
		Time - Wed Jan  4 14:35:43 UTC 2017
	Sub: issuer key ID(sub 16)(8 bytes)
		Key ID - 0x7FCC7D46ACCC4CF8
	Hash left 2 bytes - 04 0b 
	RSA m^d mod n(4096 bits) - ...
		-> PKCS-1
```

As you can see above, that 2019 expiration date was from a signature made in 2014. A subsequent 2017 signature has no expiration, so there shouldn't be problems here.

I'm going to ask upstream (the PostgreSQL Docker packagers) about this comment, since it's copy-pasted from them, but I feel safe saying just to erase the expiration part from our Dockerfile template, as it's no longer correct (and probably never was?)

Reply:
Asked about this [here](https://github.com/docker-library/postgres/issues/623)

Reply:
BTW a dev responded on that issue and confirmed the comment was stale: https://github.com/docker-library/postgres/issues/623#issuecomment-547673476

So we're good on this.